### PR TITLE
chore: rebase onto horovod 0.20.0 [DET-4225]

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
This PR includes new task containers which are build off of horovod 0.20.0.
This new version of horovod also fixes an issue that affected 
`gradient_aggregation > 1` for TF2.

## Test Plan
Added a new e2e test to test our gradient aggregation.
Ran full suite of GPU tests: https://app.circleci.com/pipelines/github/determined-ai/determined/6124/workflows/f0894812-c551-4646-8420-439f28cf2381

### CODE REVIEW

File: e2e_tests/tests/fixtures/tf_keras_grad_aggregation/grad_aggregation_test.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    multiplier: float,
+    average_aggregated_gradient: bool,
+    tf2: bool,
+) -> float:
+    idx += 1
+    if idx % aggregation_frequency == 0:
+        reduction_sum = 0.0
+        for _ in range(aggregation_frequency):
+            for rank in range(hvd.size()):
+                reduction_sum += rank * multiplier / float(hvd.size())
```
Comment (armandmcqueen) [inline e2e_tests/tests/fixtures/tf_keras_grad_aggregation/grad_aggregation_test.py:87]: This line should be expanded to make it a little easier to grok. I would do something like
```
sum_gradient_for_step = 0.0
for rank in range(hvd.size()):
  sum_gradient_for_step += rank * multiplier
avg_gradient_for_step = sum_gradient_for_step / float(hvd.size())
reduction_sum += avg_gradient_for_step
```
Or at least something like that where the logical steps are made clearer with variable names.

File: harness/determined/keras/_tf_keras_trial.py [hunk 2]
```diff
@@ -155,55 +154,6 @@ def on_train_batch_end(self, _: int, logs: Any = None) -> None:
             raise det.errors.WorkerFinishedGracefully
 
 
-class DeterminedProfiler(tf.keras.callbacks.Callback):  # type: ignore
```
Comment (armandmcqueen) [inline harness/determined/keras/_tf_keras_trial.py:158]: Why are we deleting the profiler?

File: deploy/determined_deploy/aws/templates/efs.yaml [hunk 3]
```diff
@@ -4,35 +4,35 @@ Mappings:
   RegionMap:
     ap-northeast-1:
       Master: ami-0349224eed420a3b5
-      Agent: ami-0172f5479691feb3d
+      Agent: ami-04941344de22297c2
```
Comment (armandmcqueen) [inline deploy/determined_deploy/aws/templates/efs.yaml:7]: I'm not super familiar with how we do this now. Did you go through and manually change all of these AMIs? If so that sounds like a bug in the making where we will make as mistake and break our product in a region that we don't have testing for.

**Review by aaron276h — COMMENTED** [review]
@armandmcqueen addressed your feedback. I also bumped the task containers and AMIs one more time to pull in a fix for `IndexedSlices`. Going to run e2e gpu tests as well.

**Review by armandmcqueen — APPROVED** [review]
LGTM 🚀
